### PR TITLE
Documentation 0.6.x

### PR DESCRIPTION
**Pull request by Lirt**
- Split long chapters into more documents linked from main README
- Update version 0.5.2 to 0.6.0 everywhere
- Update compatibility matrix
- Add missing information about configuration parameters or manila related information.
- Document new backup methods
- Update notes about consistency and durability

### CODE REVIEW

**Comment by Lirt** [comment]
@kayrus please review whether the updated documentation explains new features correctly.

Also, shall we not stick with the default method `method: snapshot` and set `ensureDeleted: false` (because it requires extra admin permission)?

```yaml
cinder
    method: snapshot
    ensureDeleted: "false"
    cascadeDelete: "false"
manila
    method: snapshot
    ensureDeleted: "false"
    cascadeDelete: "false"
```

File: README.md [hunk 1]
```diff
@@ -205,184 +107,35 @@ swift post -m "Temp-URL-Key:${SWIFT_TMP_URL_KEY}" my-container
...
 
-Make sure that secret `velero-credentials` exists and has proper format and content.
+Plugin supports multiple methods of creating a backup.
 
-Then install `velero` using command like this:
+Cinder backup methods:
+- **Snapshot** - Create a snapshot using Cinder.
+- **Clone** - Clone a volume using Cinder.
+- **Backup** - Create a backup using Cinder backup functionality (known in CLI as `cinder backup create`) - see [docs](https://docs.openstack.org/cinder/latest/admin/volume-backups.html).
+- **Image** - Upload a volume into Glance image service.
```
Comment (kayrus) [inline README.md:120]:
```suggestion
- **Image** - Upload a volume into Glance image service (requires  `enable_force_upload` Cinder option enabled on the server side).
```

File: README.md [hunk 2]
```diff
@@ -205,184 +107,35 @@ swift post -m "Temp-URL-Key:${SWIFT_TMP_URL_KEY}" my-container
...
-1. The snapshots are done using flag `--force`. The reason is that volumes in state `in-use` cannot be snapshotted without it (they would need to be detached in advance). In some cases this can make snapshot contents inconsistent.
-2. Snapshots in the cinder backend are not always supposed to be used as durable. In some cases for proper availability, the snapshot need to be backed up to off-site storage. Please consult if your cinder backend creates durable snapshots with your cloud provider.
+Please note two facts regarding volume backups:
+1. The snapshots are done using flag `--force`. The reason is that volumes in state `in-use` cannot be snapshotted without it (they would need to be detached in advance). In some cases this can make snapshot contents inconsistent!
+2. Durability of backups in the cinder or manila backend depend on backup method that you will use. In most cases for proper availability, the snapshot need to be backed up to off-site storage (in order to survive real datacenter incident). Please consult if chosen backup method and your cinder or manila backend setup will result in durable backups with your cloud provider.
```
Comment (kayrus) [inline README.md:130]:
```suggestion
2. Durability of backups in the Cinder or Manila backend depends on backup method that you will use. In most cases for proper availability, the snapshot needs to be backed up to off-site storage (in order to survive real datacenter incident). Please consult if chosen backup method and your Cinder or Manila backend setup will result in durable backups with your cloud provider.
```

File: docs/installation-using-cli.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    # optional snapshot method:
+    # * "snapshot" is a default cinder snapshot method
+    # * "clone" is for a full volume clone instead of a snapshot allowing the
+    # source volume to be deleted
+    # * "backup" is for a full volume backup uploaded to a Cinder backup
+    # allowing the source volume to be deleted (EXPERIMENTAL)
+    # * "image" is for a full volume backup uploaded to a Glance image
+    # allowing the source volume to be deleted (EXPERIMENTAL)
+    # requires the "enable_force_upload" Cinder option to be enabled on the server
+    method: clone
```
Comment (kayrus) [inline docs/installation-using-cli.md:58]:
```suggestion
    method: snapshot
```

File: docs/installation-using-cli.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  name: <NAME>
+  namespace: <NAMESPACE>
+spec:
+  provider: community.openstack.org/openstack-manila
+  config:
+    # optional snapshot method:
+    # * "snapshot" is a default manila snapshot method
+    # * "clone" is for a full share clone instead of a snapshot allowing the
+    # source share to be deleted
+    method: clone
```
Comment (kayrus) [inline docs/installation-using-cli.md:96]:
```suggestion
    method: snapshot
```

File: docs/installation-using-helm.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+      # optional snapshot method:
+      # * "snapshot" is a default cinder snapshot method
+      # * "clone" is for a full volume clone instead of a snapshot allowing the
+      # source volume to be deleted
+      # * "backup" is for a full volume backup uploaded to a Cinder backup
+      # allowing the source volume to be deleted (EXPERIMENTAL)
+      # * "image" is for a full volume backup uploaded to a Glance image
+      # allowing the source volume to be deleted (EXPERIMENTAL)
+      # requires the "enable_force_upload" Cinder option to be enabled on the server
+      method: clone
```
Comment (kayrus) [inline docs/installation-using-helm.md:40]:
```suggestion
      method: snapshot
```

File: docs/installation-using-helm.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+      cascadeDelete: "true"
+  # for Manila shared filesystem storage
+  - name: manila
+    provider: community.openstack.org/openstack-manila
+    config:
+      # optional snapshot method:
+      # * "snapshot" is a default manila snapshot method
+      # * "clone" is for a full share clone instead of a snapshot allowing the
+      # source share to be deleted
+      method: clone
```
Comment (kayrus) [inline docs/installation-using-helm.md:68]:
```suggestion
      method: snapshot
```

**Pull request merged**